lib/convection/model/template/resource/aws_s3_bucket.rb

```diff
@@ -50,6 +50,50 @@ def website_configuration(&block)
             properties['WebsiteConfiguration'].set(config)
           end
 
+          def full_name(terraform_import: false)
+            return "#{name.downcase}.#{stack._original_cloud}.#{stack._original_region}.rapid7.com" if terraform_import
+
+            "#{name.downcase}.#{stack.cloud}.#{stack.region}.rapid7.com"
+          end
+
+          def hcl_imports(module_path: 'root')
+            commands = ['# Run the following commands to import your infrastructure into terraform management.', '# ensure :module_path is set correctly', '']
+            module_prefix = module_path.tr('.', '-') if module_path == 'root'
+            result = {}
+
+            commands << '# Import s3 bucket and s3 bucket policy: '
+            commands << "terraform import #{module_prefix}aws_s3_bucket.#{name.downcase} #{full_name(terraform_import: true)}"
+						commands << ''
+						commands
+          end
+
+          def to_hcl_json(module_path: 'root')
+            bucket_resource = {
+              name.downcase => {
+                bucket: full_name,
+                acl: "private",
+                force_destroy: false,
+              }
+            }
+
+            data = [{aws_region: {current: {current: true } } }]
+            tf_resources = [
+              { aws_s3_bucket: bucket_resource },
+            ]
+
+            bucket_reference = "${aws_s3_bucket.#{name.downcase}.id}"
+              policy_json = resources["#{name}Policy"] && resources["#{name}Policy"].document.document.to_json.gsub(full_name, bucket_reference)
```

Comment:
1. You should generate this HCL in the `aws_s3_bucket_policy` resource class. We can't be sure that open source users (or even other users at work) are placing policies in the same stack as the bucket resource.
2. In the policy resource you can get the bucket name from the bucket attribute.